### DISCUSSION TITLE
How to specify the key name when using to_entries

### DISCUSSION THREAD

**Discussion by tommyleedberg** [mikefarah/yq#2132] [Q&A]
I'm relatively new to using the yq command and I would like to use `yq eval 'to_entries' to generate a new file from an existing one but when the new file is generated I'd like to change "key" to "name". I was able to get most of the way there I just can't figure out how to change the "key" properties name. I realize I could just do a string replace using sed but was wondering if there was a way to specify the key name in place.

I have the following:

Input file: temp.json
``` 
{
    "A": "first-letter",
    "B": "second-letter",
    "C": "third-letter"
}
```

Command:
`yq eval -P 'to_entries | {"properties": .}' temp.json | cat > temp.yaml`

This works as expected and I get the following result:

``` 
{
  "properties": [
    {
      "key": "A",
      "value": "first-letter"
    },
    {
      "key": "B",
      "value": "second-letter"
    },
    {
      "key": "C",
      "value": "third-letter"
    }
  ]
}
```

What I would like to get as an output is the following:

``` 
{
  "properties": [
    {
      "name": "A",
      "value": "first-letter"
    },
    {
      "name": "B",
      "value": "second-letter"
    },
    {
      "name": "C",
      "value": "third-letter"
    }
  ]
}
```

**Comment by mikefarah** [comment]
You can do it like this:

```
yq 'to_entries | (.[].key | key) = "name" | {"properties" : .}' examples/data1.yaml
```

The extra `(.[].key | key) = "name"` is basically saying:
- for each array element `.[]` match all the entries with a key named "key"
- This will return the value (e.g. "A") we want to update it's key, so we get  `(.[].key | key)`
- Note that `.key` is referencing a map entry, named "key" whereas "key" (without the dot) is returning the key node of the value in the map.
- Now that we have the key node, we can update it.

Hope that makes sense!
